### ISSUE TITLE
Fullscreen X11 games cannot use full resolution with 'scale 2'

### ISSUE THREAD

**Issue by lytenyn** [swaywm/sway#5444]
I am a happy user of sway on a 4k display using 'scale 2'.
However, when launching X11 games via steam, the game does not detect that it is running on a 4k screen, instead I can only select 1080p.
Playing the same games from i3 (without any scaling set) works fine.

Please tell me if any other information is required. Thank you!

sway version 1.4.

Output configuration:
output DP-1 scale 2                                                                                                                                                                                                 
output DP-1 mode 3840x2160

**Comment by emersion** [comment]
Yeah, this is a known issue, not specific to Sway. Maybe [viewporter](https://github.com/swaywm/wlroots/pull/2092) + Xwayland master will help.

**Comment by nloewen** [comment]
Is this something that should work when using XWayland master now that viewporter is implemented, or is there additional work required to make this happen.

**Comment by emersion** [comment]
I believe we'd need Xwayland master + https://github.com/swaywm/wlroots/issues/2125

**Comment by BBaoVanC** [comment]
Is there any news on this?

**Comment by emersion** [comment]
Whether `_XWAYLAND_RANDR_EMU_MONITOR_RECTS` will help will depend on the game. It'll only help for games which try to perform a modeset.

For other games, Xwayland needs to add support for fractional-scale-v1.

**Comment by BBaoVanC** [comment]
Another thing I noticed is that `xrandr` outputs my display (which is 4k, scaled 1.5x) as 1440p (but does list the preferred option of 4k as unselected) is that what `_XWAYLAND_RANDR_EMU_MONITOR_RECTS` could help solve? Or does it just mean that the game would be able to see that option of 4k to select it?

**Comment by runxiyu** [comment]
Hi, have there been any updates on this, since viewporter seems merged?

**Comment by emersion** [comment]
viewporter is not enough. Explicit support for `_XWAYLAND_RANDR_EMU_MONITOR_RECTS` is also required.
